types/atvlegacycontentkit/index.d.ts

```diff
@@ -0,0 +1,2887 @@
+// Type definitions for Apple TV ATVLegacyContentKit for TV Experience 7.7
+// Project: n/a (there is no publically available website for the SDK)
```

Comment:
there's no website we can point to that tells us what those types here are for? even if it's a website that doesn't have documentation, I think it would still be nice to have something here to provide some context

Reply:
Basically the entire SDK is private, and is generally only available to people who Apple invite to make apps for the Apple TV 2/3, so therefore there's no documentation on Apple's website about it. Basically, the entire thing came about around a decade ago when people started reverse engineering how these apps worked, and found that they used a JavaScript SDK to build them. For that reason, there isn't really any sort of formal documentation or anything like that, rather most of the information about the SDK is from other repositories on Github where people have built these apps and the first party apps themselves. So in effect there is no centralised information about this at all, and virtually no documentation (first party or otherwise) to speak of.

Much of it was also kind of discovered by me by introspecting the APIs in a web inspector and trying to map out the API surface that way. I eventually intend to make some proper documentation for building these apps, but that is still some way off, so I settled for some (hopefully) reasonably robust comments about what each part of the API does.